### ISSUE TITLE
Hunter pet still not auto casting spells.

### ISSUE THREAD

**Issue by Kwall29** [slowtorta/turtlewow-bug-tracker#4044]
**Please, provide a clear description what the bug is:** Hunter pet still not auto casting abilities sometimes the mob is half dead by me before my pet decides to growl. Most fights he sits there focus capped unless I manually click the abilities and yes the abilities are toggled for auto. It's a little frustrating since the hunters main playstyle depends on your pet using growl...

**Steps to reproduce the behavior with as much detail as possible:**

1.send pet to attack.
2. Watch pet do nothing.
3. Have to melee mob instead of shoot from afar

**IDs of affected creatures, items, quests or spells with a link to the relevant page:**

NPC: https://database.turtle-wow.org/?npc=
Item: https://database.turtle-wow.org/?item=
Quest: https://database.turtle-wow.org/?quest=
Spell: https://database.turtle-wow.org/?spell=
Object: https://database.turtle-wow.org/?object=

**Expected behavior. Describe how it should work:**
Send pet to attack shoot with bow loot dead mob.

**Comment by gh-borz** [comment]
Hard to say what the problem is, but I think the issue is pet auto-attack “swing time” is somehow off.

if you guys want to confirm this is an issue, just watch the Alexensual stream.

The problem as I see it is this:
I start shooting at same time my pet runs in. My pet pauses for a second, and let the mob run past, then starts to chase the mob that’s running to me. But because it paused for a second, it’s now out of range for its abilities, including growl, so it does not cast growl or any abilities until it gets back to me and the mob.

You can see this is happening because when you send the pet in, none of the abilities have their GCD triggered.  The problem can be overcome by manually spam clicking growl as the pet is running in; THEN it will cast growl and intercept and aggro the mob.

I originally didn’t post a bug because I thought it was because I’m playing at 380ms on average, but Alexensual has stated his ping is at ~90ms, and he still has the same problem, to the point where he’s asking if there’s something wrong with his pet, because this is not normal vanilla behaviour.

**Comment by Kwall29** [comment]
Yeah that's exactly what I'm experiencing it just makes the hunter feel extremely clunky. I haven't had this issue before on any private server or vanilla. I tried to be vocal about the issue but they keep closing tickets. I understand they have alot of issues to deal with I just expected them to fix it by now it seems to be an issue with both hunter and warlock pets but no one is talking about it. I'm glad I'm not the only one seeing the issue.
